### PR TITLE
Initial work on UpdatePFS design ADR

### LINKED ISSUE #227: Write ADR for UpdatePFS message
After our long (and fruitful) discussion about the design of the `UpdatePFS` message, we should write down the considerations into a nice ADR.

### PR DESCRIPTION
**Pull request by palango**
@Dominik1999 I started writing the `UpdatePFS`  ADR. If you find time feel free to continue, otherwise I'll finish this next week.

Fixes #227

### CODE REVIEW

**Comment by Dominik1999** [comment]
Adapted it a little bit

File: adr/001-update-pfs-message.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+* Message incluides only capacity, so the balance proofs stay private
+* Privacy is possible because possibility to broadcast only at certain thresholds and not for every update
+
+### Negative Consequences
+
+* In case where one of the channel participants is malicious, the available capacities will be computed lower than in option 1.
+
+## Detailed analysis
+
+To look at the feasibility of both design one needs to look at three different scenarios for the channel participants:
```
Comment (karlb) [inline adr/001-update-pfs-message.md]: typo: designs

File: adr/001-update-pfs-message.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+3. One of the participants is malicious.
+
+### Case 1: Two well behaving participants
+
+In this case the PFS learns about the correct channel capacities in both message designs.
+
+### Case 2: Two malicious participants
+
+In this case the PFS can not do anything to find out about the malicious behaviour. However, we assume that not all participants in the network are malicious.
+Therefore there must exist a channel, where case 3 is active.
```
Comment (karlb) [inline adr/001-update-pfs-message.md]: German word order. I suggest "Therefore, a channel must exit where case 3 is active".

File: adr/001-update-pfs-message.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+### Case 3: One malicious participant
+
+Thesis: Neither in Option 1 nor in Option 2 there is a economic incentive providing an incorrect message.
```
Comment (karlb) [inline adr/001-update-pfs-message.md]: Typo: an economic.

File: adr/001-update-pfs-message.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+Definitions: H is honest, M is malicious, H_A is deposit of H, T_H is the transferred amount of H, C_H represents the capacity of H's channel side (variables for M are equially defined)
```
Comment (karlb) [inline adr/001-update-pfs-message.md]: You probably meant `D_H` instead of `H_A`.

Comment (karlb) [inline adr/001-update-pfs-message.md]: Typo: equially

File: adr/001-update-pfs-message.md (hunk 1 above)
Comment (Dominik1999) [reply adr/001-update-pfs-message.md]: thanks, sorry for the many typos. Used the online editor :)

File: adr/001-update-pfs-message.md (hunk 2 above)
Comment (Dominik1999) [reply adr/001-update-pfs-message.md]: nice formulation

File: adr/001-update-pfs-message.md (hunk 3 above)
Comment (Dominik1999) [reply adr/001-update-pfs-message.md]: done

File: adr/001-update-pfs-message.md (hunk 4 above)
Comment (Dominik1999) [reply adr/001-update-pfs-message.md]: yes

**Comment by palango** [comment]
Thanks for improving this @Dominik1999 . I fixed some typos and reworded some sentences. Please have another look, I think we can merge this as is.

**Pull request merged**